### PR TITLE
Documentation for temporary tables

### PR DESCRIPTION
**Pull request by harshit-gangal**
Added documentation around temporary table support in Vitess

### CODE REVIEW

File: content/en/docs/reference/compatibility/mysql-compatibility.md [hunk 1]
```diff
@@ -47,6 +47,13 @@ There are further limitations to calling stored procedures using CALL:
 
 CREATE PROCEDURE is not supported. You have to create the procedure directly on the underlying MySQL servers and not through Vitess.
 
+### Temporary Tables
+
+Vitess has a limited support for temporary tables. It works only for unsharded keyspace.
```
Comment (deepthi) [inline content/en/docs/reference/compatibility/mysql-compatibility.md]:
```suggestion
Vitess has limited support for temporary tables. It works only for unsharded keyspaces.
```

File: content/en/docs/reference/compatibility/mysql-compatibility.md [hunk 2]
```diff
@@ -47,6 +47,13 @@ There are further limitations to calling stored procedures using CALL:
...
+
+If the user create a temporary table then the session will start using reserved connections for any query sent on that session
```
Comment (deepthi) [inline content/en/docs/reference/compatibility/mysql-compatibility.md]:
```suggestion
If the user creates a temporary table then the session will start using reserved connections for any query sent on that session.
```

File: content/en/docs/reference/compatibility/mysql-compatibility.md [hunk 3]
```diff
@@ -47,6 +47,13 @@ There are further limitations to calling stored procedures using CALL:
 
 CREATE PROCEDURE is not supported. You have to create the procedure directly on the underlying MySQL servers and not through Vitess.
 
+### Temporary Tables
+
+Vitess has a limited support for temporary tables. It works only for unsharded keyspace.
+
+If the user create a temporary table then the session will start using reserved connections for any query sent on that session 
+and also the plans generated by those will not be caches. But, they will continue to use the plan caches from other sessions.
```
Comment (deepthi) [inline content/en/docs/reference/compatibility/mysql-compatibility.md]:
```suggestion
The plans generated by those queries will not be cached but they will continue to use the plans cached from other sessions.  
```

**Review by deepthi — COMMENTED** [review]
Grammar and phrasing suggestions.

**Review by shlomi-noach — COMMENTED** [review]
Can you please clarify in the docs: is there a scenario where this reserved connection is finally returned to the pool, and someone else gets access to that connection, resulting in that someone able to read the temporary table?

**Comment by harshit-gangal** [comment]
I will link reserved connection to its write up

**Review by deepthi — APPROVED** [review]

**Comment by deepthi** [comment]
> Can you please clarify in the docs: is there a scenario where this reserved connection is finally returned to the pool, and someone else gets access to that connection, resulting in that someone able to read the temporary t

> I will link reserved connection to its write up

Pre-approving. You can merge after addressing @shlomi-noach's concern.

**Pull request merged**